pkg/engine/engine.go

```diff
@@ -388,23 +389,47 @@ func (e *Engine) ApplyApplicationConstraint(constraint *constraints.ApplicationC
 		if construct == nil {
 			return fmt.Errorf("construct, %s, does not exist", construct.Id())
 		}
-		new, err := e.getConstructFromInputId(constraint.ReplacementNode)
-		if err != nil {
-			return err
+		var new core.BaseConstruct
```

Comment:
`new` is a keyword in Go, probably best to pick a different name (or add more to it)